prober/utils.go

```diff
@@ -43,8 +43,6 @@ func chooseProtocol(preferredIPProtocol string, target string, registry *prometh
 	level.Info(logger).Log("msg", "Resolving target address", "preferred_ip_protocol", preferredIPProtocol)
 	resolveStart := time.Now()
 
-	defer probeDNSLookupTimeSeconds.Add(time.Since(resolveStart).Seconds())
```

Comment:
What you want to do is make this an anonymous function, so we still get the right value if the below return is hit.

Reply:
There only seems to be one valid return, the other is an error return.  Is it valid to return a DNS lookup time metric if the lookup fails?

Reply:
That's what we do elsewhere, the dns lookup still took time.

Reply:
Ok, I will implement this tomorrow.